File: src/patterns/structural/adapter/government.nim
type
    Government* = ref object


proc newGovernment*(): Government =
    return Government()

proc sendGainsInformation*(self: Government, data: tuple[taxPayerId: uint, taxPayerGains: float]): bool =
    echo "Data successfully received"
    echo "Tax Payer ID: " & $data.taxPayerId & "\nTax Payer Gains: " & $data.taxPayerGains

    return true

File: src/patterns/structural/adapter/government.go
package adapter


// std
import(
	"fmt"
)


type Government struct {}


func NewGovernment() *Government {
	return &Government{}
}

func (self *Government) SendGainsInformation(data Tuple) bool {
	fmt.Println("Data successfully received")
	fmt.Println("Tax Payer ID: ", data.TaxPayerId ,"\nTax Payer Gains: ", data.TaxPayerGains)

    return true
}